src/lavinmq/queue/queue.cr

```diff
@@ -33,6 +33,7 @@ module LavinMQ
     @consumers_lock = Mutex.new
     @message_ttl_change = Channel(Nil).new
 
+    getter basic_get_unacked = Deque(UnackedMessage).new
```

Comment:
are you removing the messages from this Dequeue when a channel closes?

Reply:
no, added removal to `close `and a spec for it now